util/src/main/java/io/kubernetes/client/pager/Pager.java

```diff
@@ -0,0 +1,107 @@
+package io.kubernetes.client.pager;
+
+import com.squareup.okhttp.Call;
+import io.kubernetes.client.ApiClient;
+import io.kubernetes.client.ApiException;
+import io.kubernetes.client.models.V1ListMeta;
+import io.kubernetes.client.util.PagerParams;
+import io.kubernetes.client.util.Reflect;
+import io.kubernetes.client.util.exception.ObjectMetaReflectException;
+import java.io.IOException;
+import java.lang.reflect.Type;
+import java.util.function.Function;
+
+public class Pager {
```

Comment:
if we don't want to make the pager shared by multiple resource types, better make it type parameterized.

```suggestion
public class Pager<ApiType, ApiListType> {
```

Reply:
added